File: python/1663/main.py
def assert_eq(a, b):
    if a == b:
      print("Correct")
    else:
      print(f"Wrong: {a} != {b}")

class Solution:
    def getSmallestString(self, n: int, k: int) -> str:
        res = [''] * n
        for i in reversed(range(0, n)):
            cur = min(26, k-(n-(n-i)))
            res[i] = chr(ord('a')+cur-1)
            k -= cur
        return ''.join(res)

assert_eq("aay", Solution().getSmallestString(3, 27))
assert_eq("aaszz", Solution().getSmallestString(5, 73))
assert_eq("vzzz", Solution().getSmallestString(4, 100))
assert_eq("zzzzz", Solution().getSmallestString(5, 130))

File: go/1663/main.go
package main

import "fmt"

func main() {
	fmt.Println(getSmallestString(3, 27))
	fmt.Println(getSmallestString(5, 73))
}

func getSmallestString(n int, k int) string {
	res := make([]byte, n)
	for i := range n {
		res[i] = 'a'
	}
	left := k - n
	for i := n - 1; i >= 0; i-- {
		if k == 0 {
			break
		}
		if left < 26 {
			res[i] = byte('a' + left)
			break
		}
		res[i] = 'z'
		left -= 'z' - 'a'
	}
	return string(res)
}
